ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessorIntegrationTest.java

```diff
@@ -0,0 +1,759 @@
+            transferTransaction1,
+            transferTransaction2);
+
+    BlockProcessingResult blockProcessingResult =
+        blockProcessor.processBlock(blockchain, worldState, blockWithTransactions);
+
+    BonsaiAccount updatedSenderAccount1 =
+        (BonsaiAccount) worldState.get(transferTransaction1.getSender());
+    BonsaiAccount updatedGenesisAccount1 =
+        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_GENESIS_1));
+    BonsaiAccount updatedGenesisAccount2 =
+        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_GENESIS_2));
+    BonsaiAccount updatedAccount0x2 =
+        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_2));
+
+    assertTrue(blockProcessingResult.isSuccessful());
+    assertThat(updatedGenesisAccount1.getBalance())
+        .isEqualTo(
+            Wei.of(
+                UInt256.fromHexString(
+                    ("0x00000000000000000000000000000000000000000000003627e8f7123739c1c8"))));
+    assertThat(updatedGenesisAccount2.getBalance())
+        .isEqualTo(
+            Wei.of(
+                UInt256.fromHexString(
+                    ("0x00000000000000000000000000000000000000000000003627e8f7123739c024"))));
+    assertThat(updatedAccount0x2.getBalance()).isEqualTo(Wei.of(2_000_000_000_000_000_000L));
+    assertThat(updatedSenderAccount1.getBalance())
+        .isLessThan(transferTransaction1Sender.getBalance());
+  }
+
+  private void processConflictedSimpleTransfersWithCoinbase(final BlockProcessor blockProcessor) {
+    // Create conflicted transactions using coinbase
+    Transaction transferTransaction1 =
+        createTransferTransaction(
+            0,
+            1_000_000_000_000_000_000L,
+            300000L,
+            5L,
+            7L,
+            ACCOUNT_2,
+            keyPair1); // ACCOUNT_GENESIS_1 -> ACCOUNT_2
+    Transaction transferTransaction2 =
+        createTransferTransaction(
+            0,
+            2_000_000_000_000_000_000L,
+            300000L,
+            5L,
+            7L,
+            coinbase.toHexString(),
+            keyPair2); // ACCOUNT_GENESIS_2 -> COINBASE
+
+    MutableWorldState worldState = worldStateArchive.getMutable();
+    BonsaiAccount transferTransaction1Sender =
+        (BonsaiAccount) worldState.get(transferTransaction1.getSender());
+    Block blockWithTransactions =
+        createBlockWithTransactions(
+            "0xd9544f389692face27352d23494dd1446d9af025067bc11b29e0eb83e258676a",
+            transferTransaction1,
+            transferTransaction2);
+
+    BlockProcessingResult blockProcessingResult =
+        blockProcessor.processBlock(blockchain, worldState, blockWithTransactions);
+
+    BonsaiAccount updatedSenderAccount1 =
+        (BonsaiAccount) worldState.get(transferTransaction1.getSender());
+
+    BonsaiAccount updatedAccount0x2 =
+        (BonsaiAccount) worldState.get(Address.fromHexStringStrict(ACCOUNT_2));
```

Comment:
This name is closer to what I was expecting - is 0x2 short for 0x000...0002?

Reply:
Yes, correct,